compiler/crates/relay-compiler/tests/relay_config_schema_json_test.rs

```diff
@@ -0,0 +1,21 @@
+use fixture_tests::assert_file_contains;
+use fixture_tests::workspace_root;
+use relay_compiler::ConfigFile;
+use schemars::gen::SchemaSettings;
+use schemars::gen::{self};
+
+#[tokio::test]
+async fn json_schema() {
+    let mut settings: SchemaSettings = Default::default();
+    settings.inline_subschemas = true;
+    let generator = gen::SchemaGenerator::from(settings);
+    let schema = generator.into_root_schema_for::<ConfigFile>();
+    let actual = serde_json::to_string_pretty(&schema).unwrap();
+    let expected = include_str!("../../../relay-compiler-config-schema.json");
+    let source_file_path = file!();
+
+    let expected_file_path = workspace_root()
+        .join(source_file_path)
+        .with_file_name("../../../relay-compiler-config-schema.json");
+    assert_file_contains(&actual, expected_file_path, expected)
```

Comment:
This reuses some of the logic from our fixture files:

1. It will fail and show a diff if the current file does not match
2. Setting `UPDATE_SNAPSHOTS=1` will cause it to update the file